Fix error when unloading kernel module (double free)

[  280.355158] ------------[ cut here ]------------
[  280.355161] refcount_t: underflow; use-after-free.
[  280.355174] WARNING: CPU: 0 PID: 1925 at lib/refcount.c:28 refcount_warn_saturate+0xc2/0x110
[  280.355183] Modules linked in: netanalyzer(O-) xt_MASQUERADE nf_conntrack_netlink xfrm_user xt_addrtype xt_mark xt_comment ip6t_REJECT nf_reject_ipv6 ip6t_rpfilter ipt_REJECT nf_reject_ipv4 xt_conntrack ebtable_nat ebtable_broute ip6table_nat ip6table_mangle ip6table_raw ip6table_security iptable_nat nf_nat iptable_mangle iptable_raw iptable_security nf_conntrack nf_defrag_ipv6 nf_defrag_ipv4 ip_set ebtable_filter ebtables ip6table_filter ip6_tables iptable_filter ip_tables x_tables nct6775 hwmon_vid i915 intel_rapl_msr iTCO_wdt intel_pmc_bxt iTCO_vendor_support wmi_bmof intel_wmi_thunderbolt intel_rapl_common x86_pkg_temp_thermal intel_powerclamp coretemp crct10dif_pclmul crct10dif_common aesni_intel crypto_simd cryptd i2c_i801 intel_spi_pci intel_spi spi_nor i2c_smbus mei_me mei igen6_edac firmware_class ttm fan thermal wmi video acpi_pad sch_fq_codel tun zram qmi_wwan zsmalloc cdc_mbim cdc_ncm cdc_ether usbnet cdc_wdm fuse configfs device_data(O) pub_key_loader(O)
[  280.355229]  [last unloaded: uio]
[  280.355231] CPU: 0 PID: 1925 Comm: modprobe Tainted: G           O      5.15.137-rt71-yocto-preempt-rt #1
[  280.355234] Hardware name: Advantech UNO-137_V2/UNO-137_V2, BIOS V1.10 06/15/2023
[  280.355235] RIP: 0010:refcount_warn_saturate+0xc2/0x110
[  280.355239] Code: 01 e8 92 41 8a 00 0f 0b 5d c3 cc cc cc cc 80 3d b2 ad 5d 01 00 75 81 48 c7 c7 28 4e 25 a1 c6 05 a2 ad 5d 01 01 e8 6e 41 8a 00 <0f> 0b 5d c3 cc cc cc cc 80 3d 8c ad 5d 01 00 0f 85 59 ff ff ff 48
[  280.355241] RSP: 0018:ffff9d3b8413fd58 EFLAGS: 00010282
[  280.355242] RAX: 0000000000000000 RBX: ffff8f2875529400 RCX: 0000000000000000
[  280.355244] RDX: 0000000000000000 RSI: ffffffffa120f289 RDI: 00000000ffffffff
[  280.355245] RBP: ffff9d3b8413fd58 R08: 0000000000000026 R09: ffff9d3b8413fcf0
[  280.355246] R10: 0000000000000028 R11: ffffffffffffffff R12: ffff8f2869ccba80
[  280.355247] R13: ffffffffc0826020 R14: ffffffffc0826098 R15: 0000000000000000
[  280.355249] FS:  00007f461f359b80(0000) GS:ffff8f29a4000000(0000) knlGS:0000000000000000
[  280.355250] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[  280.355251] CR2: 00007f66ea4735e8 CR3: 0000000129fe4000 CR4: 0000000000350ef0
[  280.355253] Call Trace:
[  280.355254]  <TASK>
[  280.355258]  ? show_regs.cold+0x1a/0x1f
[  280.355262]  ? __warn+0x80/0xd0
[  280.355265]  ? refcount_warn_saturate+0xc2/0x110
[  280.355268]  ? report_bug+0xa5/0xd0
[  280.355270]  ? handle_bug+0x40/0x70
[  280.355273]  ? exc_invalid_op+0x19/0x70
[  280.355275]  ? asm_exc_invalid_op+0x1b/0x20
[  280.355281]  ? refcount_warn_saturate+0xc2/0x110
[  280.355283]  ? refcount_warn_saturate+0xc2/0x110
[  280.355285]  kobject_put+0xc3/0xd0
[  280.355288]  netana_unregisterdevice+0x9d/0x100 [netanalyzer]
[  280.355294]  netana_pci_remove+0x23/0x70 [netanalyzer]
[  280.355298]  pci_device_remove+0x3b/0xb0
[  280.355301]  __device_release_driver+0x182/0x270
[  280.355304]  driver_detach+0xd3/0x120
[  280.355306]  bus_remove_driver+0x56/0xc0
[  280.355308]  driver_unregister+0x31/0x60
[  280.355310]  pci_unregister_driver+0x32/0x90
[  280.355312]  netana_pci_exit_module+0x10/0x392 [netanalyzer]
[  280.355318]  __do_sys_delete_module.constprop.0+0x18a/0x230
[  280.355321]  ? syscall_trace_enter.constprop.0+0x1dc/0x220
[  280.355324]  __x64_sys_delete_module+0x12/0x20
[  280.355326]  do_syscall_64+0x40/0x90
[  280.355328]  entry_SYSCALL_64_after_hwframe+0x62/0xcc
[  280.355330] RIP: 0033:0x7f461f46ca5b
[  280.355333] Code: 73 01 c3 48 8b 0d c5 33 0e 00 f7 d8 64 89 01 48 83 c8 ff c3 66 2e 0f 1f 84 00 00 00 00 00 90 f3 0f 1e fa b8 b0 00 00 00 0f 05 <48> 3d 01 f0 ff ff 73 01 c3 48 8b 0d 95 33 0e 00 f7 d8 64 89 01 48
[  280.355334] RSP: 002b:00007ffe2c591418 EFLAGS: 00000206 ORIG_RAX: 00000000000000b0
[  280.355336] RAX: ffffffffffffffda RBX: 0000000000000001 RCX: 00007f461f46ca5b
[  280.355337] RDX: 0000000000000000 RSI: 0000000000000800 RDI: 0000559117e32ce8
[  280.355338] RBP: 0000559117e32c80 R08: 0000000000000000 R09: 0000000000000000
[  280.355339] R10: 00007f461f4f8ac0 R11: 0000000000000206 R12: 0000559117e32ce8
[  280.355340] R13: 0000000000000000 R14: 0000000000000000 R15: 0000559117e324a0
[  280.355342]  </TASK>
[  280.355343] ---[ end trace 0000000000000002 ]---

diff -waurN a/netana.c b/netana.c
--- a/netana.c	2024-02-19 11:04:16.635686287 +0100
+++ b/netana.c	2024-02-26 13:10:09.142832313 +0100
@@ -839,8 +839,6 @@
         kobject_put(ptdeviceinfo->kobj_conf_dir);
         kobject_put(ptdeviceinfo->kobj_hw_dir);
         kobject_put(ptdeviceinfo->kobj_dev_info_dir);
-        kobject_put(ptdeviceinfo->kobj_dma_dir);
-        kobject_put(ptdeviceinfo->kobj_drv_info);
 
         /* delete device */
         device_destroy(netana_class->class, MKDEV(major, ptdeviceinfo->minor));
